File: src/main/webapp/js/controllers/CustomerController.js
/**
 * Created by JacksonChang408 on 6/3/2016.
 */
app.controller('CustomerController', ['$scope', '$http', function($scope, $http){


    $scope.getAllCustomers = function() {

        $http({method : 'GET',
                url : '/customer/getallcustomers'})
            .success(function(data, status) {
                $scope.customerData = data;
                $scope.status = status;
            })
            .error(function(data, status) {
                alert("Error");
            });
    };
    $scope.addCustomer = function(firstName,lastName,email,phoneNumber,source) {
        $http({method : 'PUT',
                url : '/customer/addcustomer',
             params : {"firstName": firstName,
                        "lastName": lastName,
                        "email": email,
                        "phoneNumber": phoneNumber,
                        "source": source}})
            .success(function(data, status) {
                $scope.customerData = data;
                $scope.status = status;
            })
            .error(function(data, status) {
                alert(status);
            });
    };

}]);

File: src/main/java/com/applesauce/controller/CustomerController.java
package com.applesauce.controller;

import com.applesauce.service.CustomerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import com.applesauce.model.Customer;

import java.util.List;

/**
 * Created by JacksonChang408 on 5/19/2016.
 */
@RestController
@RequestMapping("/customer")
public class CustomerController {

    @Autowired
    private CustomerService customerService;
    //private CustomerService customerService = new CustomerService();

    @RequestMapping(value = "", method = RequestMethod.GET)
    public String customerHome(){
        return "Customer Home";
    }

    @RequestMapping(value = "/test", method = RequestMethod.GET)
    public String test(){
        return "Testing mapping";
    }

    @RequestMapping(value = "/addcustomer", method = RequestMethod.PUT)
    public Customer addCustomer(@RequestParam("firstName") String firstName,
                             @RequestParam("lastName") String lastName,
                             @RequestParam("email") String email,
                             @RequestParam("phoneNumber") String phoneNumber,
                             @RequestParam("source") String source){
        return customerService.addCustomer(new Customer(firstName,lastName,email,phoneNumber,source));
    }

    @RequestMapping(value = "/getallcustomers", method = RequestMethod.GET)
    public List<Customer> getAllCustomers(){
        return customerService.getAllCustomers();
    }

    @RequestMapping(value = "/findbyfirstname/{firstName}", method = RequestMethod.GET)
    public List<Customer> findByFirstName(@PathVariable("firstName") String firstName){
        return customerService.findByFirstName(firstName);
    }

    @RequestMapping(value = "/findbylastname/{lastName}", method = RequestMethod.GET)
    public List<Customer> findByLastName(@PathVariable("lastName") String lastName){
        return customerService.findByLastName(lastName);
    }

    @RequestMapping(value = "/findbyemail/{email}", method = RequestMethod.GET)
    public Customer findByEmail(@PathVariable("email") String email){
        return customerService.findByEmail(email);
    }

}
